connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -258,6 +258,21 @@ private void commitTransaction() {
 
         long started = time.milliseconds();
 
+        if (!transactionOpen && !offsetWriter.willFlush()) {
```

Comment:
I'm wondering if the willFlush method is still necessary. Can we remove it completely and replace it with beginFlush? We don't have to worry about making sure we have an open transaction before invoking that method since it doesn't actually produce anything to Kafka.